### DISCUSSION TITLE
The docker buildx command is producing binary output.

### DISCUSSION THREAD

**Discussion by Horiodino** [docker/buildx#2720] [Q&A]
in the run hermetic build action not sure why it's giving binary output :
https://github.com/buildsafedev/examples/actions/runs/11190194394/job/31111915411

I'm using this action: https://github.com/buildsafedev/multiarch-build-action/blob/hermetic_builds/hermetic-build/action.yaml#L122

**Comment by wellkilo** [comment]
The binary output is the OCI image-layout tar stream. In the linked action, the command combines --push with --output type=oci but does not specify dest. An OCI exporter writes its archive to the destination; when no file destination is provided, it can go to stdout, which GitHub Actions displays as binary noise.

Choose one export mode:

1. If the goal is to push the image, remove the OCI output:

       docker buildx build \
         --tag "$IMAGE:$TAG" \
         --platform="$PLATFORM" \
         --attest type=provenance,mode=min \
         --push .

2. If the goal is an OCI-layout artifact, do not stream it to the log; give it a file destination:

       docker buildx build \
         --platform="$PLATFORM" \
         --output type=oci,dest=image.oci.tar .

       # upload image.oci.tar as an Actions artifact

3. If you need both, declare two explicit outputs (registry and OCI file), subject to the builder/exporter version supporting multiple outputs.

The problematic lines are 115-117 / 129-131 of the linked action: --push followed by --output type=oci. This is output bytes by design, not corrupted log encoding. Redirecting stdout to /dev/null would hide it but also discard the artifact; dest=... is the correct fix.
